dev/PushNotifications/PushNotifications.idl

```diff
@@ -0,0 +1,133 @@
+        Guid TaskClsid{ get; };
+
+        PushNotificationRegistrationOption Option{ get; };
+
+        // The conditions under which Push Triggers would execute
+        Windows.ApplicationModel.Background.IBackgroundCondition[] GetConditions();
+        void SetConditions(Windows.ApplicationModel.Background.IBackgroundCondition[] conditions);
+    };
+
+    enum PushNotificationChannelStatus
+    {
+        InProgress, // The request is in progress and there is no retry operation
+        InProgressRetry, // The request is in progress and is in a backoff retry state. Check ExtendedError for HRESULT for retryable error.
+        CompletedSuccess, // The request completed successfully
+        CompletedFailure, // The request failed with some critical internal error. Check ExtendedError for HRESULT
+    };
+
+    // The PushNotificationChannel Progress result
+    struct PushNotificationCreateChannelStatus
+    {
+        // Either InProgress or InProgressRetry status
+        PushNotificationChannelStatus status;
+
+        // The last extended error we failed Channel requests on that caused the inprogress retry status. E_PENDING if this is just progress status.
+        HRESULT extendedError;
+
+        // Total Retries so far
+        UInt32 retryCount;
+    };
+
+    [experimental]
+    runtimeclass PushNotificationChannel
+    {
+        PushNotificationChannel(Windows.Networking.PushNotifications.PushNotificationChannel channel);
+
+        // The Channel Uri for app to Post a notification to.
+        Windows.Foundation.Uri Uri { get; };
+
+        // Expiration of the Channel
+        Windows.Foundation.DateTime ExpirationTime { get; };
+
+        // Unsubscribes the channel
+        void Close();
+
+        // Event handler for Push Notifications
+        event Windows.Foundation.TypedEventHandler<PushNotificationChannel, PushNotificationReceivedEventArgs> PushReceived;
+    }
+
+    [experimental]
+    runtimeclass PushNotificationCreateChannelResult
+    {
+        PushNotificationCreateChannelResult(
+            PushNotificationChannel channel,
+            HRESULT extendedError,
+            PushNotificationChannelStatus status);
+
+        // The Push channel associated with the Result. Valid only if status is CompletedSuccess.
+        PushNotificationChannel Channel { get; };
+
+        // More detailed error code in addition to the ChannelStatus state.
+        HRESULT ExtendedError{ get; };
+
+        // The Status of the CreateChannel operation
+        PushNotificationChannelStatus Status { get; };
+    };
+
+    [experimental]
+    runtimeclass PushNotificationRegistrationToken
+    {
+        PushNotificationRegistrationToken(
+            UInt64 cookie,
+            Windows.ApplicationModel.Background.IBackgroundTaskRegistration taskRegistration);
+
+        // The cookie from CoRegisterClassObject
+        UInt64 Cookie{ get; };
+
+        // The Registration token for the Push Trigger
```

Comment:
```suggestion
        // The Registration for the Push Trigger
```

The enclosing class is the RegistrationToken. This member is the background task Registration.